### PR TITLE
{bio}[foss/2016b] GROMACS v2016.3 (REVIEW)

### PR DESCRIPTION
**Pull request by jrbosch**
I selected toolchain foss2016 and not foss2017 because CUDA doesn't support GCC 6.3 at its latest version.

The NVML library location depends on driver version. On my system the NVML library is located in /usr/lib64/nvidia. Therefore, I include it in patch file that modifies FindNVML.cmake.

### CODE REVIEW

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# * Fotis Georgatos <fotis@cern.ch>
+# * George Tsouloupas <g.tsouloupas@cyi.ac.cy>
+# * Kenneth Hoste <kenneth.hoste@ugent.be>
+# * Adam Huffman <adam.huffman@crick.ac.uk>
+# License::   MIT/GPL
+##
+
+name = 'GROMACS'
+version = '2016.3'
+versionsuffix = '-GPU-enabled-build'
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: @jrbosch could we have a shorter `versionsuffix` here?

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles.
+
+This is a GPU enabled build, containing both MPI and threadMPI builds.
+"""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'openmp': True, 'usempi': True}
+
+source_urls = ['ftp://ftp.gromacs.org/pub/gromacs/']
+sources = [SOURCELOWER_TAR_GZ]
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: Please add an SHA256 checksum.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+]
+
+dependencies = [
+    ('CUDA', '8.0.61_375.26'),
+]
+
+builddependencies = [
+    ('CMake', '3.7.2'),
+]
+
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: Please add a sanity check.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3_amend_search_for_nvml_lib.patch [hunk 4]
```diff
@@ -8,7 +8,7 @@ diff -ru gromacs-2016.orig/cmake/FindNVML.cmake gromacs-2016/cmake/FindNVML.cmak
    # passed to the installer, or the root where they later found the
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3_amend_search_for_nvml_lib.patch]: Please include an explanation at the top as to why you've changed the original patch

**Comment by verdurin** [comment]
Ping @jrbosch ?

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb (hunk 3 above)
Comment (jrbosch) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: With the easyblock sanity check not enough ??

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3_amend_search_for_nvml_lib.patch (hunk 4 above)
Comment (jrbosch) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3_amend_search_for_nvml_lib.patch]: done!

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb (hunk 2 above)
Comment (jrbosch) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: done

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb (hunk 1 above)
Comment (jrbosch) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: done!

**Comment by verdurin** [comment]
@jrbosch thanks for the updates. I'm seeing a `307` error when my test build tries to download the sources. Will try again later in the day.

Regarding the sanity check, that was my initial over-zealousness as a new maintainer. I always check the easyblock now...

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
ca005.camp.thecrick.org - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2459f5a35cb4d26bee01b4b865b7adaa for a full test report.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb (hunk 3 above)
Comment (verdurin) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-foss-2016b-GPU-enabled-build.eb]: As I said elsewhere, ignore this. My error.

**Comment by verdurin** [comment]
Merging, thanks @jrbosch

**Pull request merged**